src/ophyd_async/core/_device/device_save_loader.py

```diff
@@ -0,0 +1,122 @@
+from enum import Enum
+from typing import Any, Dict, List
+
+import yaml
+from bluesky import Msg
+from numpy import ndarray
+
+from ophyd_async.core import Device, SignalRW
+
+
+def get_signal_RWs_from_device(
+    device: Device, path_prefix: str = "", signalRWs: Dict[str, SignalRW] = {}
+) -> Dict[str, SignalRW]:
+    """
+    Get all the SignalRWs from a device and store them with their dotted attribute
+    paths. Used by the save and load methods.
+
+    Parameters
+    ----------
+    device : Device
+        Ophyd device to retrieve read-write signals from.
+
+    path_prefix : str
+        For internal use, leave blank when calling the method.
+
+    SignalRWs : dict
+        A dictionary matching the string attribute path of a SignalRW with the
+        signal itself. Leave blank when calling the method.
+
+    Returns
+    -------
+    SignalRWs : dict
+        A dictionary matching the string attribute path of a SignalRW with the
+        signal itself.
+    """
+
+    for attr_name, attr in device.children():
+        dot = ""
+        # Place a dot inbetween the upper and lower class.
+        # Don't do this for highest level class.
+        if path_prefix:
+            dot = "."
+        dot_path = f"{path_prefix}{dot}{attr_name}"
+        if type(attr) is SignalRW:
+            signalRWs[dot_path] = attr
+        get_signal_RWs_from_device(attr, path_prefix=dot_path)
+    return signalRWs
```

Comment:
Using a mutable default arg will cause problems:
https://web.archive.org/web/20200221224620id_/http://effbot.org/zone/default-values.htm

I suggest you return a dict and update with the child devices.

```suggestion
    signals: Dict[str, SignalRW] = {}
    for attr_name, attr in device.children():
        dot_path = f"{path_prefix}{attr_name}"
        if isinstance(attr, SignalRW):
            signals[dot_path] = attr
        attr_signals = get_signal_RWs_from_device(attr, path_prefix = dot_path + ".")
        signals.update(attr_signals)
    return signals
```